django-stubs/core/cache/backends/base.pyi

```diff
@@ -20,50 +20,50 @@ class BaseCache:
+        self, key: Any, default: Any | None, timeout: float | None = ..., version: Any | None = ...
     ) -> Any | None: ...
-    def has_key(self, key: Any, version: int | None = ...) -> bool: ...
-    async def ahas_key(self, key: Any, version: int | None = ...) -> bool: ...
-    def incr(self, key: Any, delta: int = ..., version: int | None = ...) -> int: ...
-    async def aincr(self, key: Any, delta: int = ..., version: int | None = ...) -> int: ...
-    def decr(self, key: Any, delta: int = ..., version: int | None = ...) -> int: ...
-    async def adecr(self, key: Any, delta: int = ..., version: int | None = ...) -> int: ...
+    def has_key(self, key: Any, version: Any | None = ...) -> bool: ...
+    async def ahas_key(self, key: Any, version: Any | None = ...) -> bool: ...
+    def incr(self, key: Any, delta: int = ..., version: Any | None = ...) -> int: ...
+    async def aincr(self, key: Any, delta: int = ..., version: Any | None = ...) -> int: ...
+    def decr(self, key: Any, delta: int = ..., version: Any | None = ...) -> int: ...
+    async def adecr(self, key: Any, delta: int = ..., version: Any | None = ...) -> int: ...
     def __contains__(self, key: Any) -> bool: ...
-    def set_many(self, data: dict[Any, Any], timeout: float | None = ..., version: int | None = ...) -> list[Any]: ...
+    def set_many(self, data: dict[Any, Any], timeout: float | None = ..., version: Any | None = ...) -> list[Any]: ...
     async def aset_many(
-        self, data: dict[Any, Any], timeout: float | None = ..., version: int | None = ...
+        self, data: dict[Any, Any], timeout: float | None = ..., version: Any | None = ...
     ) -> list[Any]: ...
-    def delete_many(self, keys: Iterable[Any], version: int | None = ...) -> None: ...
-    async def adelete_many(self, keys: Iterable[Any], version: int | None = ...) -> None: ...
+    def delete_many(self, keys: Iterable[Any], version: Any | None = ...) -> None: ...
+    async def adelete_many(self, keys: Iterable[Any], version: Any | None = ...) -> None: ...
     def clear(self) -> None: ...
     async def aclear(self) -> None: ...
-    def incr_version(self, key: Any, delta: int = ..., version: int | None = ...) -> int: ...
-    async def aincr_version(self, key: Any, delta: int = ..., version: int | None = ...) -> int: ...
-    def decr_version(self, key: Any, delta: int = ..., version: int | None = ...) -> int: ...
-    async def adecr_version(self, key: Any, delta: int = ..., version: int | None = ...) -> int: ...
+    def incr_version(self, key: Any, delta: int = ..., version: Any | None = ...) -> int: ...
+    async def aincr_version(self, key: Any, delta: int = ..., version: Any | None = ...) -> int: ...
+    def decr_version(self, key: Any, delta: int = ..., version: Any | None = ...) -> int: ...
+    async def adecr_version(self, key: Any, delta: int = ..., version: Any | None = ...) -> int: ...
```

Comment:
If `version` argument is set to `Any`, I think increment/decrement should return `Any`(?) and `delta` should perhaps be the same type as `version`.

It sure looks like there could be a `TypeVar` here?

Reply:
I'm not totally sure about this but since Django e.g. adds `delta` to `version` at the end of `incr_version` like:

```python
return version + delta
```

Correct me if I'm wrong, but I suppose there could be a case of declaring something like

```python
def incr_version(self, key: Any, delta: SupportsAdd[T, T] = ..., version: SupportsAdd[T, T] | None = ...) -> T: ...
```

`SupportsAdd` is declared in the `_typeshed` module and can be imported under `if TYPE_CHECKING:`

Reply:
I guess with decr_version existing this can't work as written

so this patch is probably dead